doc/modules/compose.rst

```diff
@@ -426,21 +447,21 @@ By default, the remaining rating columns are ignored (``remainder='drop'``)::
   >>> from sklearn.feature_extraction.text import CountVectorizer
   >>> from sklearn.preprocessing import OneHotEncoder
   >>> column_trans = ColumnTransformer(
-  ...     [('city_category', OneHotEncoder(dtype='int'),['city']),
+  ...     [('categories', OneHotEncoder(dtype='int'), ['city']),
   ...      ('title_bow', CountVectorizer(), 'title')],
   ...     remainder='drop')
 
   >>> column_trans.fit(X)
-  ColumnTransformer(transformers=[('city_category', OneHotEncoder(dtype='int'),
+  ColumnTransformer(transformers=[('categories', OneHotEncoder(dtype='int'),
                                    ['city']),
                                   ('title_bow', CountVectorizer(), 'title')])
 
-  >>> column_trans.get_feature_names()
-  ['city_category__x0_London', 'city_category__x0_Paris', 'city_category__x0_Sallisaw',
-  'title_bow__bow', 'title_bow__feast', 'title_bow__grapes', 'title_bow__his',
-  'title_bow__how', 'title_bow__last', 'title_bow__learned', 'title_bow__moveable',
-  'title_bow__of', 'title_bow__the', 'title_bow__trick', 'title_bow__watson',
-  'title_bow__wrath']
+  >>> column_trans.get_feature_names_out()
+  ['categories__city_London', 'categories__city_Paris',
+   'categories__city_Sallisaw', 'title_bow__bow', 'title_bow__feast',
+   'title_bow__grapes', 'title_bow__his', 'title_bow__how', 'title_bow__last',
+   'title_bow__learned', 'title_bow__moveable', 'title_bow__of', 'title_bow__the',
+   'title_bow__trick', 'title_bow__watson', 'title_bow__wrath']
```

Comment:
I changed a line above for a cosmit and as a result it hides an important discussion that was happening here. Sorry for that.

The discussion was about: shall we add a new constructor parameter to skip the verbose `categories__` prefix in this doctest when there is no risk of colliding feature names? It could be named:

- `prefix_feature_names="when_colliding"` (default)
- `prefix_feature_names="always"`

If we do so, we need to amend the following paragraph of SLEP7:

- https://github.com/scikit-learn/enhancement_proposals/blob/master/slep007/proposal.rst#meta-estimators

> ColumnTransformer by default adds a prefix to the output feature names, indicating the name of the transformer applied to them. If a column is in the output as a part of passthrough, it won't be prefixed since no operation has been applied on it.

I am in favor of making the change now to avoid problems with backward compat if we want to do this change later.

Reply:
+1 on adding it now, I'm not entirely sure about the default +0.5 on that, I guess? In particular it means that adding transformers can change the names of existing features. The typical use-case where you wouldn't want prefixes is where you have separate feature types like categorical, continuous and bow, like here?